### ISSUE TITLE
contenteditable ARIA mapping silently skipped during spec generation

### ISSUE THREAD

**Issue by saphryn** [markuplint/markuplint#3896]
## Bug Description

The `contenteditable` attribute is silently skipped during ARIA spec generation in `aria.ts` (lines 478-482). The attribute has defined implicit ARIA semantics in the HTML-ARIA specification, but the generated spec data contains no mapping for it. This means markuplint has no validation rule for the ARIA implications of `contenteditable`.

## Root Cause

In `packages/@markuplint/html-spec/generator/aria.ts`, lines 478-482:

```ts
const htmlAttrName = $($implicitProp).find('th:nth-of-type(1) a').eq(0).text();
if (htmlAttrName === 'contenteditable') {
  // FIXME: The contenteditable attribute's ARIA semantics depend on ancestor elements
  // (an element inherits contenteditable from its parent). Evaluating this requires
  // cross-element ancestor traversal which is not available during spec generation.
  continue;
}
```

The FIXME documents the reason: `contenteditable` is an inherited attribute. An element's effective editable state depends on whether any ancestor has `contenteditable="false"` or `contenteditable=""` (or `="true"`), plus whether there's an intervening `contenteditable="inherit"` or explicit override. The spec generator operates on per-element data with no ancestor context, so it skips the attribute entirely rather than producing an incorrect mapping.

However, this means:
1. The HTML-ARIA implicit property mapping for `contenteditable` is **absent** from the generated spec
2. Any downstream validation of ARIA semantics for editable content is **impossible**
3. The skip is silent — there is no warning or documentation for users about this gap

## Impact

- `contenteditable=""` (or `="true"`) makes an element and its descendants editable — this communicates editable state to assistive technology via ARIA semantics
- `contenteditable="false"` explicitly removes editability
- `contenteditable="inherit"` restores inheritance from the parent
- Without this mapping, code that relies on markuplint's spec data cannot reason about:
  - Whether an editable element is communicating its state correctly
  - Whether `contenteditable` is used in a valid structure
  - Whether the effective ARIA role is consistent with the editable state

## Code Location

`packages/@markuplint/html-spec/generator/aria.ts` lines 478-482

## Recommended Fix

1. **Model the inheritance** in the generated spec as a property-level note or constraint on `contenteditable` values, even if the cross-element check must happen at rule-validation time rather than spec-generation time
2. **Add a lint rule** that validates `contenteditable` ARIA implications at document analysis time (when the full DOM tree is available)
3. **At minimum**, document the gap in the spec output or add a warning counter so the omission is not silent

**Comment by YusukeHirao** [comment]
Triaged. The skip is intentional, as the FIXME in `generator/aria.ts` documents: `contenteditable`'s effective state depends on ancestor inheritance, which cannot be resolved during per-element spec generation. Producing a context-free mapping would be wrong more often than producing none.

Your recommendation 2 is the technically correct direction: the ARIA implications of `contenteditable` should be evaluated at **rule-evaluation time**, when the full DOM tree is available and the effective editable state can be computed by walking ancestors. That is new-rule-scale work rather than a generator fix.

Worth noting for expectation-setting: the HTML-ARIA implicit property mapping for `contenteditable` is the `aria-readonly`-related mapping, so the practical accessibility impact of the current gap is limited.

**Decision: keeping this open as an enhancement** (rule-time evaluation of `contenteditable` ARIA semantics). No near-term implementation is planned; it may also be folded into the v6 spec/rule design (#3178).
